include/ros_alarms/listener.hpp

```diff
@@ -94,11 +110,25 @@ ::AlarmListener(ros::NodeHandle &nh, std::string alarm_name)
 {
   // Service to query alarm server
   __get_alarm = __nh.serviceClient<ros_alarms::AlarmGet>("/alarm/get");
-  __get_alarm.waitForExistence();
-
+  bool service_exists = __get_alarm.waitForExistence(ros::Duration(2.0));
+  if(!service_exists)
+  {
+    __ok = false;
+    std::string msg = __PRETTY_FUNCTION__;
+    msg += ": timed out waiting for service /alarm/get";
+    ROS_WARN(msg.c_str());
+  }
 
   // Subscribes to list of recently modified alarms from alarm server
-  __update_subscriber = __nh.subscribe("/alarm/updates", 10, &AlarmListener<callable_t>::__alarm_update, this);
+  try
+  {
+    __update_subscriber = __nh.subscribe("/alarm/updates", 10, &AlarmListener<callable_t>::__alarm_update, this);
+  }
+  catch(ros::Exception &e)
+  {
+    __ok = false;
+    ROS_WARN((std::string(__PRETTY_FUNCTION__) + ": " + e.what()).c_str());
```

Comment:
How it is right now, the user can call the ok() method after the constructor and decide if he wants to kill the node or what he want's to do. I'm thinking that's flexibility you would want to give the user.